### DISCUSSION TITLE
bunch.app freezes (beachballs) for many seconds opening first bunch after launch

### DISCUSSION THREAD

**Discussion by bunnyhero** [ttscoff/bunch-docs#84] [Q&A]
has anyone else experienced this? i'd never seen this till the big rewrite, and it only happens to me with the first bunch after startup.

steps to reproduce:

1. launch bunch.app
2. click on bunch menu and open a bunch (doesn't matter which)

result: bunch menu icon turns yellow for a good 10 seconds at least, while nothing else happens. clicking on the menu when it's in this state turns the pointer into the spinning rainbow pinwheel.

after about 10 seconds or so, all of the apps in the bunch finally start launching.

3. click on bunch menu and open any other bunch

result: no delay, everything opens and runs as expected.

this is with the preference "launch items in parallel" turned off, by the way.
 
i turned on debug logging to see if there was anything interesting there. in this example, i opened the "Comms" bunch at 21:04:30. the app froze, not actually launching anything, until 21:04:40 ("Generating status menu")

```
2021-06-23 21:04:30: ⚪[Comms] Open >>>>>>>>>>>>>>>>>>>>>
2021-06-23 21:04:30: ⚪[Comms] Launching app (Calendar)
2021-06-23 21:04:30: ⚪[Comms] Launching app (Notes)
2021-06-23 21:04:30: ⚪[Comms] Launching app (Reminders)
2021-06-23 21:04:30: ⚪[Comms] Launching app (Discord)
2021-06-23 21:04:30: ⚪[Comms] Launching app (Messages)
2021-06-23 21:04:30: ⚪[Comms] Launching app (Messenger)
2021-06-23 21:04:30: ⚪[Comms] Launching app (Mail.app)
2021-06-23 21:04:40: 🔘Generating status menu
```

any ideas on how to troubleshoot this would be great. thanks!

**Comment by ttscoff** [comment]
I haven't seen this at all. I know you said it happens no matter which 
Bunch you click, but could I see what's in your Comms bunch just so I 
can try to test locally?

-Brett

**Reply by bunnyhero** [reply]
yeah sure! it's very simple. i've pasted it verbatim below:

```
Calendar
Notes
Reminders
Discord
Messages
Messenger
Mail.app
```

thanks for taking a look. i wonder if it's something else i have installed-- let me know if i should send you anything else!

**Reply by clarke** [reply]
I am seeing this same behavior on both M1 and Intel Macs, though my delay on both platforms are closer to 60 seconds. Happy to provide information and/or help diagnosing, as well.

**Reply by ttscoff** [reply]
@clarke please clarify that your symptoms match the OP, i.e. delay on the first launch, but fine after that.

@bunnyhero and @clarke Please let me know what OS version you're on.

**Reply by clarke** [reply]
I am seeing the same behavior with the delay happening on first launch of a bunch, and then subsequent launches of that same bunch being quick.

Two more symptoms I have observed yesterday and today.
1. Each bunch (I have 3) is slow when it is launched for the first time after the Bunch application has initially launched, and is then quick on subsequent launches as long as the Bunch application remains running.
2. On a 2018 Intel MacBook Pro 13" running yesterday, the initial triggering of a bunch caused a long delay and the Bunch application quit. I tried multiple times throughout the day and was unable to get any bunches to launch successfully.

My setups that I've tested are:
* 2020 M1 Mac mini running Big Sur
* 2018 Intel MacBook Pro 13" running Catalina

**Reply by ttscoff** [reply]
Next thing I need to know is whether the folks on this thread are running "Bunch" or "Bunch Beta." I'm unable to replicate the issue with either build, and as of right now they should be identical, but I'd like to know if the issue is specific to one or the other.

**Comment by rueshannbrantley** [comment]
I am having the same issue with the beachball when launching any bunch. In the example below, the first app Drafts launches immediately, but there is a delay of 100 seconds and then all of the other apps appear. The preference "launch items in parallel" is turned off. This bunch launched with no delays on the version prior to 1.4.

Here is a copy of the logging:

    2021-06-25 09:29:58: ⚪[H] Open >>>>>>>>>>>>>>>>>>>>>
    2021-06-25 09:29:58: ⚪[H] Launching app (Drafts)
    2021-06-25 09:30:08: ⚪[H] Launching app (OmniFocus)
    2021-06-25 09:30:18: ⚪[H] Launching app (DEVONthink 3)
    2021-06-25 09:30:28: ⚪[H] Launching app (Safari)
    2021-06-25 09:30:38: ⚪[H] Launching app (Obsidian)
    2021-06-25 09:30:48: ⚪[H] Launching app (Microsoft Teams)
    2021-06-25 09:30:58: ⚪[H] Launching app (Microsoft Outlook)
    2021-06-25 09:31:08: ⚪[H] Launching app (Microsoft Edge)
    2021-06-25 09:31:18: ⚪[H] Launching app (Path Finder)
    2021-06-25 09:31:28: ⚪[H] Launching app (Timery)
    2021-06-25 09:31:38: ⚪[H] Running AppleScript (tell application "Moom" to arrange windows according to snapshot "H")

Here is a copy of that bunch

    Drafts
    OmniFocus
    DEVONthink 3
    Safari
    Obsidian
    Microsoft Teams
    Microsoft Outlook
    Microsoft Edge
    Path Finder
    Timery
    * tell application "Moom" to arrange windows according to snapshot "KH"

**Reply by ttscoff** [reply]
The OP is about this happening on first launch, and then everything launching fine after. Is that the case here, or are you seeing the delay every time. And is it just this specific Bunch?

**Comment by ttscoff** [comment]
Could everybody on this thread try deleting their preferences and starting fresh? It sounds like you all jumped from the 1.3 version to the final release of 1.4 (i.e. not part of the beta). There may be an issue I didn't catch with updating the existing preferences.

1. Quit Bunch
2. Locate and trash the file `~/Library/Preferences/com.brettterpstra.Bunch.plist`. You can do this from Terminal with `rm ~/Library/Preferences/com.brettterpstra.Bunch.plist`, or in Finder by hitting ⌘+⇧+G, pasting in the path, then dragging the selected file to the Trash.
3. Reboot your machine to make sure the preferences cache gets cleared
4. Relaunch Bunch and see if the problem still persists. You'll need to configure your preferences again, but there aren't many, so this shouldn't be much of a hassle

**Reply by clarke** [reply]
I just tried the above on my M1 Mac mini and the result is the same. There is a long pause when activating a bunch the first time, and then finally the bunch launches as expected.

I've been running the Bunch beta on this computer for multiple weeks and only began encountering this issue earlier this week.

**Reply by clarke** [reply]
I just launched the Bunch binary from the command line to see if I could get any errors to print and it worked without any delay. I ran:

`$ /Applications/Bunch\ Beta.app/Contents/MacOS/Bunch\ Beta`

Once it was running like that (the terminal command remained in the foreground), my bunches are starting immediately, as they did before.

**Reply by ttscoff** [reply]
I can't imagine why that would make a difference...

**Comment by rueshannbrantley** [comment]
I am having the same issue after the deletion of the plist/reboot. The issue occurs just on the first bunch I run and doesn’t matter which bunch I use. After the first bunch runs, all of the bunches afterwards runs quickly.

From: Brett Terpstra ***@***.***>
Date: Friday, June 25, 2021 at 12:13 PM
To: ttscoff/bunch ***@***.***>
Cc: Rueshann Brantley ***@***.***>, Comment ***@***.***>
Subject: Re: [ttscoff/bunch] bunch.app freezes (beachballs) for many seconds opening first bunch after launch (#84)

Could everybody on this thread try deleting their preferences and starting fresh? It sounds like you all jumped from the 1.3 version to the final release of 1.4 (i.e. not part of the beta). There may be an issue I didn't catch with updating the existing preferences.

  1.  Quit Bunch
  2.  Locate and trash the file ~/Library/Preferences/com.brettterpstra.Bunch.plist. You can do this from Terminal with rm ~/Library/Preferences/com.brettterpstra.Bunch.plist, or in Finder by hitting ⌘+⇧+G, pasting in the path, then dragging the selected file to the Trash.
  3.  Reboot your machine to make sure the preferences cache gets cleared
  4.  Relaunch Bunch and see if the problem still persists. You'll need to configure your preferences again, but there aren't many, so this shouldn't be much of a hassle

**Comment by rueshannbrantley** [comment]
I am running “Bunch.”

From: Brett Terpstra ***@***.***>
Date: Friday, June 25, 2021 at 4:12 PM
To: ttscoff/bunch ***@***.***>
Cc: Rueshann Brantley ***@***.***>, Comment ***@***.***>
Subject: Re: [ttscoff/bunch] bunch.app freezes (beachballs) for many seconds opening first bunch after launch (#84)

Next thing I need to know is whether the folks on this thread are running "Bunch" or "Bunch Beta." I'm unable to replicate the issue with either build, and as of right now they should be identical, but I'd like to know if the issue is specific to one or the other.

**Comment by ttscoff** [comment]
I've thus far been unable to replicate this issue. It's clearly affecting a number of users, so I'd like to figure out what's happening.

I need to figure out what's different between my setup and everyone else in this thread. When you post new info, please be sure to include your OS version, Bunch version, type of Mac (model, Intel, M1) and any other info you think might be relevant. The more info the better, as I've been unsuccessful in making this happen on my own machine, whether testing with my usual setup or creating entirely new Bunch folders to test with.

**Comment by rueshannbrantley** [comment]
MacBook Pro (16-inch, 2019)
Model Name:     MacBook Pro
Model Identifier:              MacBookPro16,1
Processor Name:              8-Core Intel Core i9
MacOS 11.4
Bunch 1.4.1

From: Brett Terpstra ***@***.***>
Date: Monday, June 28, 2021 at 3:49 PM
To: ttscoff/bunch ***@***.***>
Cc: Rueshann Brantley ***@***.***>, Comment ***@***.***>
Subject: Re: [ttscoff/bunch] bunch.app freezes (beachballs) for many seconds opening first bunch after launch (#84)

I've thus far been unable to replicate this issue. It's clearly affecting a number of users, so I'd like to figure out what's happening.

I need to figure out what's different between my setup and everyone else in this thread. When you post new info, please be sure to include your OS version, Bunch version, type of Mac (model, Intel, M1) and any other info you think might be relevant. The more info the better, as I've been unsuccessful in making this happen on my own machine, whether testing with my usual setup or creating entirely new Bunch folders to test with.

**Comment by clarke** [comment]
I did some debugging this morning, since I'm still observing the significant delays on all three of my regular computers. Here is the information about the specific computer used for debugging today:

OS version: macOS Big Sur 11.5
Bunch version: 1.4.4 (118) (Bunch Beta)
Type of Mac: Mac mini (M1, 2020)


The consistent behavior I am observing is 10 seconds between the launch of each application when activating a bunch the first time after launching the Bunch app. The steps I take to consistently reproduce the issue are:

1. Open Bunch
2. Launch bunch using Bunch's menubar menu

After the initial triggering of any bunch, all subsequent activations of any bunch work as expected.

I removed all of my regular bunches and created 2 test bunches with Apple-provided default applications to remove the possibility that I had a problematic application. The behavior was the same.

Here are the debug logs from 2 runs. The first shows launching Bunch, opening a bunch, closing that same bunch, and then subsequently launching a second bunch. There is a 10 second delay between the launch of the applications in the first bunch, after which the other closing and opening of bunches happens immediately.

The second log shows launching Bunch, opening a bunch, opening a second bunch, and then closing the two open bunches one at a time. The same result was observed as above, specifically around the 10 second delay in-between application launches from the first bunch triggered after the launch of the application.

I will continue to try additional debugging with what I have access to and will post any new information if I find it.

Run 1:
```
2021-07-29 07:16:57: 🔘Added file logger (/Users/XXXX/Library/Logs/Bunch Beta/com.brettterpstra.BunchBeta 2021-07-28--12-48-37-638.log)
2021-07-29 07:16:57: ⚪Launched ==========================================
2021-07-29 07:16:57: ⚪[Test] Setting display title to "Test"
2021-07-29 07:16:57: ⚪[Test] Setting display title to "Test"
2021-07-29 07:16:57: ⚪[Test 2] Setting display title to "Test 2"
2021-07-29 07:16:57: ⚪[Test 2] Setting display title to "Test 2"
2021-07-29 07:16:57: 🔘Observing file:///Users/clarke/bunches/ for changes
2021-07-29 07:16:57: 🔘Generating status menu
2021-07-29 07:17:03: ⚪[Test] Open >>>>>>>>>>>>>>>>>>>>>
2021-07-29 07:17:03: ⚪[Test] Launching app (Safari)
2021-07-29 07:17:13: ⚪[Test] Launching app (Terminal)
2021-07-29 07:17:23: ⚪[Test] Launching app (TextEdit)
2021-07-29 07:17:33: 🔘Generating status menu
2021-07-29 07:17:43: ⚪[Test] Closing <<<<<<<<<<<<<<<<<<
2021-07-29 07:17:44: ⚪[Test] Quitting app Safari
2021-07-29 07:17:44: ⚪[Test] Quitting app Terminal
2021-07-29 07:17:44: ⚪[Test] Quitting app TextEdit
2021-07-29 07:17:44: 🔘Generating status menu
2021-07-29 07:17:49: ⚪[Test 2] Open >>>>>>>>>>>>>>>>>>>>>
2021-07-29 07:17:49: ⚪[Test 2] Launching app (System Preferences)
2021-07-29 07:17:49: ⚪[Test 2] Launching app (TV)
2021-07-29 07:17:49: ⚪[Test 2] Launching app (App Store)
2021-07-29 07:17:49: 🔘Generating status menu
2021-07-29 07:17:53: ⚪[Test 2] Closing <<<<<<<<<<<<<<<<<<
2021-07-29 07:17:53: ⚪[Test 2] Quitting app System Preferences
2021-07-29 07:17:53: ⚪[Test 2] Quitting app TV
2021-07-29 07:17:53: ⚪[Test 2] Quitting app App Store
2021-07-29 07:17:53: 🔘Generating status menu
2021-07-29 07:17:56: ⚪[Test] Open >>>>>>>>>>>>>>>>>>>>>
2021-07-29 07:17:56: ⚪[Test] Launching app (Safari)
2021-07-29 07:17:56: ⚪[Test] Launching app (Terminal)
2021-07-29 07:17:56: ⚪[Test] Launching app (TextEdit)
2021-07-29 07:17:56: 🔘Generating status menu
2021-07-29 07:18:00: ⚪[Test] Closing <<<<<<<<<<<<<<<<<<
2021-07-29 07:18:00: ⚪[Test] Quitting app Safari
2021-07-29 07:18:00: ⚪[Test] Quitting app Terminal
2021-07-29 07:18:00: ⚪[Test] Quitting app TextEdit
2021-07-29 07:18:00: 🔘Generating status menu
```

Run 2:
```
2021-07-29 07:18:46: 🔘Added file logger (/Users/XXXX/Library/Logs/Bunch Beta/com.brettterpstra.BunchBeta 2021-07-28--12-48-37-638.log)
2021-07-29 07:18:46: ⚪Launched ==========================================
2021-07-29 07:18:46: ⚪[Test] Setting display title to "Test"
2021-07-29 07:18:46: ⚪[Test] Setting display title to "Test"
2021-07-29 07:18:46: ⚪[Test 2] Setting display title to "Test 2"
2021-07-29 07:18:46: ⚪[Test 2] Setting display title to "Test 2"
2021-07-29 07:18:46: 🔘Observing file:///Users/clarke/bunches/ for changes
2021-07-29 07:18:46: 🔘Generating status menu
2021-07-29 07:18:55: ⚪[Test] Open >>>>>>>>>>>>>>>>>>>>>
2021-07-29 07:18:55: ⚪[Test] Launching app (Safari)
2021-07-29 07:19:05: ⚪[Test] Launching app (Terminal)
2021-07-29 07:19:15: ⚪[Test] Launching app (TextEdit)
2021-07-29 07:19:25: 🔘Generating status menu
2021-07-29 07:19:33: ⚪[Test 2] Open >>>>>>>>>>>>>>>>>>>>>
2021-07-29 07:19:33: ⚪[Test 2] Launching app (System Preferences)
2021-07-29 07:19:33: ⚪[Test 2] Launching app (TV)
2021-07-29 07:19:33: ⚪[Test 2] Launching app (App Store)
2021-07-29 07:19:33: 🔘Generating status menu
2021-07-29 07:19:42: ⚪[Test] Closing <<<<<<<<<<<<<<<<<<
2021-07-29 07:19:42: ⚪[Test] Quitting app Safari
2021-07-29 07:19:42: ⚪[Test] Quitting app Terminal
2021-07-29 07:19:42: ⚪[Test] Quitting app TextEdit
2021-07-29 07:19:42: 🔘Generating status menu
2021-07-29 07:19:46: ⚪[Test 2] Closing <<<<<<<<<<<<<<<<<<
2021-07-29 07:19:46: ⚪[Test 2] Quitting app System Preferences
2021-07-29 07:19:46: ⚪[Test 2] Quitting app TV
2021-07-29 07:19:46: ⚪[Test 2] Quitting app App Store
2021-07-29 07:19:46: 🔘Generating status menu
2021-07-29 07:19:50: 🔘Removed directory observer
2021-07-29 07:19:50: ⚪Quitting ------------------------------------------
```

**Reply by ttscoff** [reply]
Thanks for the info, still trying to replicate. Will keep you posted if I find anything over the weekend.

**Comment by ihatemornings** [comment]
I’m seeing this too, exactly as described: on the first run after powering up, Bunch launches the first app in the list immediately (Fantastical) then the logs show 10 seconds between each subsequent app launch (but what I actually see is the orange Bunch icon for 70 seconds then all the remaining apps launch at once (or in very quick succession).

- Bunch Beta 1.4.10 (161)
- macOS 12.4
- MacBook Pro (14-inch, 2021)
- M1 chip
- Ethernet and audio interface via USB-C hub
- external monitor via HDMI
- Launch Items In Parallel not enabled

```
2022-08-15 11:57:03: ⚪[Bandcamp] Open >>>>>>>>>>>>>>>>>>>>>
2022-08-15 11:57:03: ⚪[Bandcamp] Hiding all apps
2022-08-15 11:57:03: ⚪[Bandcamp] Launching app (Fantastical)
2022-08-15 11:57:13: ⚪[Bandcamp] Launching app (Slack)
2022-08-15 11:57:23: ⚪[Bandcamp] Launching app (Sublime Text)
2022-08-15 11:57:33: ⚪[Bandcamp] Launching app (Drafts)
2022-08-15 11:57:43: ⚪[Bandcamp] Launching app (nvUltra)
2022-08-15 11:57:53: ⚪[Bandcamp] Launching app (Obsidian)
2022-08-15 11:58:03: ⚪[Bandcamp] Launching app (iTerm)
2022-08-15 11:58:13: ⚪[Bandcamp] Running Shell Script (/var/folders/fk/vd1l328107g4qbw01j0k7ywr0000gn/T/TemporaryItems/NSIRD_Bunch\ Beta_XI7Em7/0A7F7532-F97A-4CCB-83BE-87B71F5202E9.sh)
2022-08-15 11:58:13: ⚪[Bandcamp] Focusing app (Slack)
2022-08-15 11:58:13: 🔘Generating status menu
2022-08-15 11:58:13: 🔘Generating status menu
```

**Comment by esamecar** [comment]
Still the same here with my M1 Mac. I know that the Hook.app had the same problems and they fixed it by compiling with a special/different option or something ... It took them a while to figure out what was wrong. (Sorry, I am no programmer and therefore have no clue.)

**Reply by lpb2ha** [reply]
[Hook productivity mac app](https://itunes.apple.com/WebObjects/MZStore.woa/wa/viewSoftware?id=740251957) dev here. Hook uses AppleScript to connect with some apps. If an app with which Hook is trying to communicate is unresponsive, Hook gives up after a certain period of time. This may be relevant: [Activating apps via AppleScript sometimes... Apple Community](https://discussions.apple.com/thread/8631074).

If an app integrates with another via AppleScript and that other app is seriously misbehaving, the user might think it's an issue with the original app.

**Reply by esamecar** [reply]
I had a similar "beachballing" / delay effect with Hook over a year ago. Somehow it was fixed. I meant this discussion, but most probably those issues are not related to each other - but again it seems to be related to intel vs. non-intel macs: https://discourse.hookproductivity.com/t/fixed-with-3-0-compiled-universal-m1-intel-preview-app-not-working-for-some-m1-big-sur-customers/2889/33

**Comment by ttscoff** [comment]
The perfect 10s intervals in the log are very weird. I'll ping the Hook 
devs and see if their issue was related.

**Comment by aikenadam** [comment]
> has anyone else experienced this? i'd never seen this till the big rewrite, and it only happens to me with the first bunch after startup.
> 
> steps to reproduce:
> 
> 1. launch bunch.app
> 2. click on bunch menu and open a bunch (doesn't matter which)
> 
> result: bunch menu icon turns yellow for a good 10 seconds at least, while nothing else happens. clicking on the menu when it's in this state turns the pointer into the spinning rainbow pinwheel.
> 
> after about 10 seconds or so, all of the apps in the bunch finally start launching.
> 
> 3. click on bunch menu and open any other bunch
> 
> result: no delay, everything opens and runs as expected.
> 
> this is with the preference "launch items in parallel" turned off, by the way.
> 
> i turned on debug logging to see if there was anything interesting there. in this example, i opened the "Comms" bunch at 21:04:30. the app froze, not actually launching anything, until 21:04:40 ("Generating status menu")
> 
> ```
> 2021-06-23 21:04:30: ⚪[Comms] Open >>>>>>>>>>>>>>>>>>>>>
> 2021-06-23 21:04:30: ⚪[Comms] Launching app (Calendar)
> 2021-06-23 21:04:30: ⚪[Comms] Launching app (Notes)
> 2021-06-23 21:04:30: ⚪[Comms] Launching app (Reminders)
> 2021-06-23 21:04:30: ⚪[Comms] Launching app (Discord)
> 2021-06-23 21:04:30: ⚪[Comms] Launching app (Messages)
> 2021-06-23 21:04:30: ⚪[Comms] Launching app (Messenger)
> 2021-06-23 21:04:30: ⚪[Comms] Launching app (Mail.app)
> 2021-06-23 21:04:40: 🔘Generating status menu
> ```
> 
> any ideas on how to troubleshoot this would be great. thanks!

I had exactly the same issue. It took me a while to solve it, and in my case it was a bit trivial and 
it is  embarrassing to admit...
I use a mac with silicon, just to reference..
It turned out that despite Buch showed in preference that it has full access to the disc ( Accessibility enabled )
it did not have a full disc access... It had only access to the folders Documents Folder and Desktop Folder.
Full Disk Access wasn't granted and after changing that in Settings, Bunch works with no issue at all.
Maybe it is coincidence that in my case symptoms was the same but anyway..

**Reply by ihatemornings** [reply]
I was hoping this would solve it, but giving Bunch full disk access in System Preferences > Security & Privacy doesn’t eliminate the issue for me. :(

**Reply by aikenadam** [reply]
Unfortunately, when I restarted my computer the issue returned. Beta version below fixed it.

**Comment by ttscoff** [comment]
I haven't figured out why yet, but I'm fairly certain a semaphore is timing out when Bunches are launched sequentially. I'm going to guess that everyone on this thread has parallel launching disabled in Preferences? If you enable that, does the issue go away?

**Comment by clarke** [comment]
I enabled parallel launching and the issue does not happen, so I’ve left it on.

**Comment by ttscoff** [comment]
Ok, I think I have this solved in the most recent beta (available now). It required removing some of the checks that keep sequential launches in line, so things launch in very rapid succession. 

I may need to revisit how that was being handled, but the semaphore that kept the next launch from happening until the current one completed was never being triggered by a callback block, and so it would just wait until timeout (10s). And the next app would then block the thread until IT timed out, and so on down the line. So if you were launching 6 apps, there would be a 60 second pause while all the semaphores timed out. I can't explain why this only happens the first time after launch, that doesn't make any sense. But for now I've just removed the code that causes it to wait for the completion block to finish and it seems to work pretty well.

As this will change the way some sequential-launch Bunches behave, please help me out by testing your Bunches and seeing if there's any new weird behavior. Leave "Launch Items in Parallel" disabled, as this change only affects sequential launching.

You can grab the beta at <https://bunchapp.co/download/>

**Reply by ihatemornings** [reply]
I installed beta 166 and I’m still seeing the same delay. 😔

**Reply by ihatemornings** [reply]
But “launch in parallel” works great for me, so I’m happy using that!

**Reply by ttscoff** [reply]
@ihatemornings can you show me the bunch causing the delays? Even if it's every Bunch you have, show me one that definitely does it.

**Reply by kimonostereo** [reply]
The beta fixed this issue for me. Thanks so much for the fix!

**Reply by aikenadam** [reply]
I can confirm beta fixed the issue for me as well. :) THX.

**Reply by ihatemornings** [reply]
@ttscoff I reproed the issue with beta 166, then saw there was a 167 and installed that. Now the issue seems to be fixed, but I get a new crash when I try to open a bunch containing some zsh shell scripts. I’ll narrow down that repro case and report it as a separate bug, but I think the beach ball delay issue is fixed for me too!
